diffrax/integrate.py

```diff
@@ -801,8 +1132,38 @@ def _promote(yi):
         else:
             tnext = t0 + dt0
     tnext = jnp.minimum(tnext, t1)
+
+    if delays is not None:
+        delays_fn = [
+            lambda t, y, args, fn=fn: fn(t, y, args) * direction for fn in delays.delays
```

Comment:
`delays.delays` may now be a PyTree, rather than an iterable.
(And so `delays_fn` should be one as well.)

Reply:
yup